pyne/data.pyx

```diff
@@ -944,10 +944,8 @@ def branch_ratio(from_nuc, to_nuc, use_metastable=True):
         from_nuc = pyne.nucname.id(from_nuc)
         to_nuc = pyne.nucname.id(to_nuc)
         ms = from_nuc % 10000
-        from_nuc = (from_nuc // 10000) * 10000
```

Comment:
Why is this wrong?  If this isn't working, shouldn't it be fixed in `metastable_id()`?

Reply:
Yeah, this is confusing. `metastable_id()` actually does this `(from_nuc // 10000) * 10000` calculation already. `metastable_id()` looks up the id from the nuclear data, and if it isn't found it returns the nuclide id supplied.  Duplicating the `(from_nuc // 10000) * 10000` calculation here ensures that you get the ground state. 

This is bad for computing branch ratios such as `Se86 -> Br86m`.  Br86m doesn't exist, so the the branch ratio should be 0.0.  However, the `(from_nuc // 10000) * 10000` calculation here was making it so that we would actually get the branch ratio for `Se86 -> Br86`, which is 1.0.

Reply:
Got it.  So now it has me thinking through the permutations of how `metastable_id()` works:
* user provides: ground state nuc_id & m = 0: returns ground state nuc_id
* user provides: ground state nuc_id & m > 0:
     * finds state m: returns metastable nuc_id
     * does not find state m: returns ground state nuc_id
* user provides: metastable nuc_id & m =0: returns ground state nuc_id [1]
* user provides: metastable nuc_id & m > 0:
     * finds state m: returns metastable nuc_id
     * does not find state m: returns original metastable nuc_id [2]

Two issues/questions for this last case:
[1] do we want to lose the originally provided metastable state?  This is not wrong, per se, but a behavior that may be non-intuitive to some: using the `mestastable_id()` method to extract the ground state from a metastable nuc_id.  Should this be documented?

[2] what if m does not match the originally provided metastable state?  This will still return a nuc_id for a metastable state, but not the one requested, and possibly not a valid metastable state.  It may appear to have found a metastable state since it returns a non-ground state id.

Reply:
I agree that [1] is not wrong per se, and should definitely be documented. 

[2] is weird and was what was causing the issue seen here. So I think the question is what should happen if a metastable state is invalid / doesn't exist but is requested.  Should this throw and error? Should it return an invalid value (such as -1)? Or should we maintain the current behaviour? I feel like all of these have benefits and drawbacks...

Reply:
OK - After more thinking about this, I went ahead and implemented `metastable_id()` returning -1 in C++ and None in Python. This allows the other functions to handle the case of non-existent isomers in the same way that they would handle any other non-existent nuclide (which is also implemented here).